---
title: "metabric.Rmd"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(imsig)

```

Update 4th August 2020

# ImgSig package to interrogate METABRIC dataset

Immune cell signature signatures for profiling the microenvironment of solid tumours https://cancerimmunolres.aacrjournals.org/content/6/11/1388

The genomic and transcriptomic architecture of 2,000 breast tumours reveals novel subgroups (METABRIC) https://www.nature.com/articles/nature10983

Should be able to download METABRIC dataset from European Genome-Phenome Archive (http://www.ebi.ac.uk/ega/) EGAS00000000083, but site seems to be down for now.

Managed to download from cBioPortal

ImSig documentation https://github.com/ajitjohnson/imsig

```{r read in files}

clinical_sample <-read.delim("data_clinical_sample.txt", header=TRUE)

clinical_patient <- read.delim("data_clinical_patient.txt", header=TRUE)

expression_median <- read.delim("data_expression_median.txt", header=TRUE)

gene_matrix <-  read.delim("data_gene_matrix.txt", header=TRUE)

mrna_median_all_sample_zscores <- read.delim("data_mRNA_median_all_sample_Zscores.txt", header=TRUE)

mrna_median_zscores <-  read.delim("data_mRNA_median_Zscores.txt", header=TRUE)

## important ones to use: clinical_patient for survival information and mrna_median_all_sample_zscores for gene matrix ##


```


```{r preparing files for imsig}

# preparing expression data for ImSig

exp <- mrna_median_all_sample_zscores

# remove EntrezID
exp[2] <- NULL

# preparing clinical data - keep only patient ID, overall survival in months and status (live/dead)

cli <- clinical_patient[,c(1, 13, 14)]

# remove top 4 rows (descriptive)

cli <- cli[-c(1,2,3, 4),]

# change column headings

names(cli)[names(cli) == "Overall.Survival..Months."] <- "time"

names(cli)[names(cli) == "Overall.Survival.Status"] <- "status"

# make patient IDs row names and remove column

rownames(cli) <- cli[,1]

cli[1] <- NULL

```

