---
title: "MCF_Peaks"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(stringr)
library(org.Hs.eg.db)
library(clusterProfiler)
```

## MLLT1 ChIP-Seq: MCF-7

Previously - aligned .fastq files to .bam, removed duplicates. Used Homer to find peaks in DMSO samples, then annotate peaks using DMSO + SGC files. Used factor style and histone style to find peaks.

```{r 7 days}
# read in files
all_data_7d_factor <- read.delim("MCF_MLLT1_PEAKS_7D_FACTOR.quant.txt", header = TRUE, sep = "\t")
all_data_7d_histone <- read.delim("MCF_MLLT1_PEAKS_7D_HISTONE.quant.txt", header = TRUE, sep = "\t")

# select specific columns and rearrange

peaks_7d_factor <- all_data_7d_factor[,c(6, 8, 16:20)]
peaks_7d_factor <- subset(peaks_7d_factor, select=c(3, 1, 7, 2, 4, 5, 6))
names(peaks_7d_factor)[names(peaks_7d_factor) == "Peak.Score"] <- "DMSO.Peak.Score"
names(peaks_7d_factor)[names(peaks_7d_factor) == "MCF_SGC_7D_IP..Tag.Count.in.given.bp..3176761.0.Total..normalization.factor...3.15..effective.total...10000000."] <- "SGC.Peak.Score"

peaks_7d_histone <- all_data_7d_histone[,c(6, 8, 16:20)]
peaks_7d_histone <- subset(peaks_7d_histone, select=c(3, 1, 7, 2, 4, 5, 6))
names(peaks_7d_histone)[names(peaks_7d_histone) == "Peak.Score"] <- "DMSO.Peak.Score"
names(peaks_7d_histone)[names(peaks_7d_histone) == "MCF_SGC_7D_IP..Tag.Count.in.given.bp..3176761.0.Total..normalization.factor...3.15..effective.total...10000000."] <- "SGC.Peak.Score"


# style factor: calculating fold change then log2 transform

peaks_7d_factor$DMSO.Peak.Score <- peaks_7d_factor$DMSO.Peak.Score + 1
peaks_7d_factor$SGC.Peak.Score <- peaks_7d_factor$SGC.Peak.Score + 1

peaks_7d_factor$LFC <- peaks_7d_factor$SGC.Peak.Score/peaks_7d_factor$DMSO.Peak.Score
peaks_7d_factor <- subset(peaks_7d_factor, select=c(1:3, 8, 4:7))
peaks_7d_factor$LFC <- log2(peaks_7d_factor$LFC)

# select only protein-coding genes

peaks_7d_factor<-peaks_7d_factor[!(peaks_7d_factor$Gene.Type=="ncRNA"),]
peaks_7d_factor<-peaks_7d_factor[!(peaks_7d_factor$Gene.Type=="pseudo"),]

# select only promoter binding regions
# peaks_7d_factor <- peaks_7d_factor[str_detect(peaks_7d_factor$Annotation, "promoter"), ]

write.csv(peaks_7d_factor, "peaks_7d_factor.csv")


# style histone: calculating fold change then log2 transform

peaks_7d_histone$DMSO.Peak.Score <- peaks_7d_histone$DMSO.Peak.Score + 1
peaks_7d_histone$SGC.Peak.Score <- peaks_7d_histone$SGC.Peak.Score + 1

peaks_7d_histone$LFC <- peaks_7d_histone$SGC.Peak.Score/peaks_7d_histone$DMSO.Peak.Score
peaks_7d_histone <- subset(peaks_7d_histone, select=c(1:3, 8, 4:7))
peaks_7d_histone$LFC <- log2(peaks_7d_histone$LFC)

# select only protein-coding genes
peaks_7d_histone<-peaks_7d_histone[!(peaks_7d_histone$Gene.Type=="ncRNA"),]
peaks_7d_histone<-peaks_7d_histone[!(peaks_7d_histone$Gene.Type=="pseudo"),]

# select only promoter binding regions
# peaks_7d_histone <- peaks_7d_histone[str_detect(peaks_7d_histone$Annotation, "promoter"), ]

write.csv(peaks_7d_histone, "peaks_7d_histone.csv")

```

```{r 24 hours}
# read in files
all_data_24h_factor <- read.delim("MCF_MLLT1_PEAKS_24H_FACTOR.quant.txt", header = TRUE, sep = "\t")
all_data_24h_histone <- read.delim("MCF_MLLT1_PEAKS_24H_HISTONE.quant.txt", header = TRUE, sep = "\t")

# select specific columns and rearrange

peaks_24h_factor <- all_data_24h_factor[,c(6, 8, 16:20)]
peaks_24h_factor <- subset(peaks_24h_factor, select=c(3, 1, 7, 2, 4, 5, 6))
names(peaks_24h_factor)[names(peaks_24h_factor) == "Peak.Score"] <- "DMSO.Peak.Score"
names(peaks_24h_factor)[names(peaks_24h_factor) == "MCF_SGC_24H_IP..Tag.Count.in.given.bp..2782430.5.Total..normalization.factor...3.59..effective.total...10000000."] <- "SGC.Peak.Score"

peaks_24h_histone <- all_data_24h_histone[,c(6, 8, 16:20)]
peaks_24h_histone <- subset(peaks_24h_histone, select=c(3, 1, 7, 2, 4, 5, 6))
names(peaks_24h_histone)[names(peaks_24h_histone) == "Peak.Score"] <- "DMSO.Peak.Score"
names(peaks_24h_histone)[names(peaks_24h_histone) == "MCF_SGC_24H_IP..Tag.Count.in.given.bp..2782430.5.Total..normalization.factor...3.59..effective.total...10000000."] <- "SGC.Peak.Score"


# style factor: calculating fold change then log2 transform

peaks_24h_factor$DMSO.Peak.Score <- peaks_24h_factor$DMSO.Peak.Score + 1
peaks_24h_factor$SGC.Peak.Score <- peaks_24h_factor$SGC.Peak.Score + 1

peaks_24h_factor$LFC <- peaks_24h_factor$SGC.Peak.Score/peaks_24h_factor$DMSO.Peak.Score
peaks_24h_factor <- subset(peaks_24h_factor, select=c(1:3, 8, 4:7))
peaks_24h_factor$LFC <- log2(peaks_24h_factor$LFC)

# select only protein-coding genes

peaks_24h_factor<-peaks_24h_factor[!(peaks_24h_factor$Gene.Type=="ncRNA"),]
peaks_24h_factor<-peaks_24h_factor[!(peaks_24h_factor$Gene.Type=="pseudo"),]

# select only promoter binding regions
# peaks_24h_factor <- peaks_24h_factor[str_detect(peaks_24h_factor$Annotation, "promoter"), ]

write.csv(peaks_24h_factor, "peaks_24h_factor.csv")


# style histone: calculating fold change then log2 transform

peaks_24h_histone$DMSO.Peak.Score <- peaks_24h_histone$DMSO.Peak.Score + 1
peaks_24h_histone$SGC.Peak.Score <- peaks_24h_histone$SGC.Peak.Score + 1

peaks_24h_histone$LFC <- peaks_24h_histone$SGC.Peak.Score/peaks_24h_histone$DMSO.Peak.Score
peaks_24h_histone <- subset(peaks_24h_histone, select=c(1:3, 8, 4:7))
peaks_24h_histone$LFC <- log2(peaks_24h_histone$LFC)

# select only protein-coding genes
peaks_24h_histone<-peaks_24h_histone[!(peaks_24h_histone$Gene.Type=="ncRNA"),]
peaks_24h_histone<-peaks_24h_histone[!(peaks_24h_histone$Gene.Type=="pseudo"),]

# select only promoter binding regions
# peaks_24h_histone <- peaks_24h_histone[str_detect(peaks_24h_histone$Annotation, "promoter"), ]

write.csv(peaks_24h_histone, "peaks_24h_histone.csv")

```

```{r GO}
entrez_id <- all_data_7d_factor[,c(6, 8, 12, 16:20)]
entrez_id <- subset(entrez_id, select=c(4, 3, 1, 8, 2, 5, 6, 7))
names(entrez_id)[names(entrez_id) == "Peak.Score"] <- "DMSO.Peak.Score"
names(entrez_id)[names(entrez_id) == "MCF_SGC_7D_IP..Tag.Count.in.given.bp..3176761.0.Total..normalization.factor...3.15..effective.total...10000000."] <- "SGC.Peak.Score"

entrez_id$DMSO.Peak.Score <- entrez_id$DMSO.Peak.Score + 1
entrez_id$SGC.Peak.Score <- entrez_id$SGC.Peak.Score + 1

entrez_id$LFC <- entrez_id$SGC.Peak.Score/entrez_id$DMSO.Peak.Score
entrez_id <- subset(entrez_id, select=c(1:4, 9, 5:8))
entrez_id$LFC <- log2(entrez_id$LFC)

# select only protein-coding genes

entrez_id<-entrez_id[!(entrez_id$Gene.Type=="ncRNA"),]
entrez_id<-entrez_id[!(entrez_id$Gene.Type=="pseudo"),]

# Keep only rows from table without NAs
entrez_id <- entrez_id[is.na(entrez_id$Entrez.ID)==FALSE,]
# Remove duplicated entries
entrez_id <- entrez_id[!duplicated(entrez_id$Entrez.ID),]

# select only peaks that decrease upon addition of inhibitor
decrease <- entrez_id[(entrez_id$LFC < 0),]


# Change Entrez IDs from numbers to characters
geneset <- as.character(decrease$Entrez.ID)

# This will take a little while to run
ego <- enrichGO(gene = geneset, 
                universe = NULL, #all available genes in database
                OrgDb = org.Hs.eg.db, #Hs: homo sapiens
                ont ="BP", #molecular function, biological process, cellular component
                pAdjustMethod = "BH",
                pvalueCutoff = 0.01,
                qvalueCutoff = 0.05,  #q value is FDR adjusted p value
                readable = TRUE) #will show gene symbol in images later rather than Entrez Gene ID

# dimensions - number of GO terms
dim(ego)

simp_ego <- simplify(ego)

dim(simp_ego)

# copy and paste table into Excel
mcf_7d_GOgenes <- data.frame(simp_ego$ID, simp_ego$Description, simp_ego$p.adjust, simp_ego$geneID)

# select only peaks that increase upon addition of inhibitor
increase <- entrez_id[(entrez_id$LFC > 0),]

```


