---
title: "ChIP-Seq metaplot"
output:
  word_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(data.table)
library(stringr)
library(plyr)
library(dplyr)
library(cowplot)
library(tidyverse)
library(gridExtra)
library(ggplot2)
library(ggpubr)
```

## MLLT1 ChIP-Seq in MCF-7 and SUM159 breast cancer cell lines
Generating metaplot to visualise changes in distribution of MLLT1 over peaks when treated with SGC-iMLLT compared to DMSO

Text files below are not original Homer output, but must have modified dataframes and then saved as new files - original Homer output in /stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/ (same file names)

```{r}
# read in files
sum_24h <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/SUM_MLLT1_24H_average.txt", sep=",")
sum_7d <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/SUM_MLLT1_7D_average.txt", sep=",")
mcf_24h <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/MCF_MLLT1_24H_average.txt", sep=",")
mcf_7d <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/SUM_MLLT1_7D_average.txt", sep=",")


p1 <- ggplot(sum_24h, aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 peaks in SUM159 with 24 hours treatment",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted")


p2 <- ggplot(sum_7d, aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 peaks in SUM159 with 7 days treatment",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted")

p3 <- ggplot(mcf_24h, aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 peaks in MCF-7 with 24 hours treatment",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted")

p4 <- ggplot(mcf_7d, aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 peaks in MCF-7 with 7 days treatment",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted")

```

```{r, fig.height=10, fig.width = 16}

grid.arrange(p1, p2, p3, p4, nrow=2)

```


## MLLT1 ChIP-Seq and CRISPR + SGCiMLLT RNA-Seq histogram
Generating metaplot to visualise changes in distribution of MLLT1 over peaks when treated with SGC-iMLLT compared to DMSO

Only looking specifically at peaks from genes that were down or upregulated from combined RNA-Seq results

Not sure when modified original files, but need to get them into ggplot format!


```{r}
# read in files
sum_24h_down <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/SUM_MLLT1_24H_down_degs_tss.txt", sep="\t")
  
sum_24h_up <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/SUM_MLLT1_24H_up_degs_tss.txt", sep="\t")

sum_7d_down <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/SUM_MLLT1_7D_down_degs_tss.txt", sep="\t")

sum_7d_up <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/SUM_MLLT1_7D_up_degs_tss.txt", sep="\t")

mcf_24h_down <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/MCF_MLLT1_24H_down_degs_tss.txt", sep="\t")

mcf_24h_up <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/MCF_MLLT1_24H_up_degs_tss.txt", sep="\t")

mcf_7d_down <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/MCF_MLLT1_7D_down_degs_tss.txt", sep="\t")

mcf_7d_up <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/MCF_MLLT1_7D_up_degs_tss.txt", sep="\t")

```

```{r}
# get Homer output into ggplot format - tidy up all 8 dataframes in one go

list <- list(sum_24h_up, sum_24h_down, sum_7d_up, sum_7d_down,
          mcf_24h_up, mcf_24h_down, mcf_7d_up, mcf_7d_down)

list <- lapply(list, function(x){
  colnames(x)[1] <- "position"
  y <- x[,c(1, 2)]
  colnames(y)[2] <- "count"
  y$treatment <- "DMSO"
  z <- x[,c(1, 5)]
  colnames(z)[2] <- "count"
  z$treatment <- "SGC-iMLLT"
  x <- rbind(y, z)
    x
} )


```



```{r}
p1 <- ggplot(list[[1]], aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 upregulated genes in SUM159 + 24h iMLLT",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme(plot.title = element_text(size=12), 
        axis.title = element_text(size=10))

p2 <- ggplot(list[[2]], aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 downregulated genes in SUM159 + 24h iMLLT",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme(plot.title = element_text(size=12), 
        axis.title = element_text(size=10))


p3 <- ggplot(list[[3]], aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 upregulated genes in SUM159 + 7d iMLLT",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme(plot.title = element_text(size=12), 
        axis.title = element_text(size=10))

p4 <- ggplot(list[[4]], aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 downregulated genes in SUM159 + 7d iMLLT",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme(plot.title = element_text(size=12), 
        axis.title = element_text(size=10))

p5 <- ggplot(list[[5]], aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 upregulated genes in MCF-7 + 24h iMLLT",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme(plot.title = element_text(size=12), 
        axis.title = element_text(size=10))

p6 <- ggplot(list[[6]], aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 downregulated genes in MCF-7 + 24h iMLLT",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme(plot.title = element_text(size=12), 
        axis.title = element_text(size=10))


p7 <- ggplot(list[[7]], aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 upregulated genes in MCF-7 + 7d iMLLT",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted")+
  theme(plot.title = element_text(size=12), 
        axis.title = element_text(size=10))

p8 <- ggplot(list[[8]], aes(x=position, y=count, color=treatment)) +
  geom_line(size=2, aes(linetype=treatment)) +
  labs(title="MLLT1 distribution at MLLT1 downregulated genes in MCF-7 + 7d iMLLT",
       x="Position relative to peak centre (bp)", 
       y = "Read count (per 10 million reads") +
  scale_color_manual(values=c("grey", "#55A0FF")) + theme_classic() +
  scale_linetype_manual(values = c("dashed", "solid")) + 
  geom_vline(xintercept = 0, linetype="dotted") +
  theme(plot.title = element_text(size=12), 
        axis.title = element_text(size=10))

```

```{r, fig.height=10, fig.width = 16}

grid.arrange(p1, p3, p2, p4, nrow=2)

grid.arrange(p5, p7, p6, p8, nrow=2)

```


## MLLT1 ChIP-Seq and CRISPR + SGCiMLLT RNA-Seq boxplots
Generating metaplot to visualise changes in distribution of MLLT1 over peaks when treated with SGC-iMLLT compared to DMSO

Only looking specifically at peaks from genes that were down or upregulated from combined RNA-Seq results

Need to calculate LFC from counts and plot this, then can do stats


```{r}
# read in files
sum_24h_down <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/SUM_MLLT1_24H_down_degs_bp_tss.txt", sep="\t")
  
sum_24h_up <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/SUM_MLLT1_24H_up_degs_bp_tss.txt", sep="\t")

sum_7d_down <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/SUM_MLLT1_7D_down_degs_bp_tss.txt", sep="\t")

sum_7d_up <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/SUM_MLLT1_7D_up_degs_bp_tss.txt", sep="\t")

mcf_24h_down <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/MCF_MLLT1_24H_down_degs_bp_tss.txt", sep="\t")

mcf_24h_up <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/MCF_MLLT1_24H_up_degs_bp_tss.txt", sep="\t")

mcf_7d_down <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/MCF_MLLT1_7D_down_degs_bp_tss.txt", sep="\t")

mcf_7d_up <- read.delim("/stopgap/donglab/ling/t1-data/MLLT1/ChIP-Seq/analyses/results/MCF_MLLT1_7D_up_degs_bp_tss.txt", sep="\t")

```

```{r}
# get Homer output into ggplot format

list <- list(sum_24h_up, sum_24h_down, sum_7d_up, sum_7d_down,
          mcf_24h_up, mcf_24h_down, mcf_7d_up, mcf_7d_down)

list <- lapply(list, function(x){
  lfc <- as.vector(log2(x[21]/x[20]))
  x$lfc <- lfc
  x$lfc <- as.vector(unlist(x$lfc))
  x$timepoint <- ifelse(grepl("24", colnames(x)[20]), "24 hours",
                        ifelse(grepl("7", colnames(x)[20]), "7 days", ""))
  x <- x[,-c(1:21)]
  x
    } )

sum_up <- rbind(list[[1]], list[[3]])
sum_down <- rbind(list[[2]], list[[4]])
mcf_up <- rbind(list[[5]], list[[7]])
mcf_down <- rbind(list[[6]], list[[8]])

```

```{r}
my_list <- list(c("24 hours", "7 days"))

p1 <- ggplot(sum_up, aes(x=timepoint, y=lfc)) +
   geom_jitter(width=0.25, size = 0.05, aes(colour = timepoint)) +
  geom_boxplot(width=0.5, outlier.shape=NA, alpha = 0.2) +
  labs(title="SUM159 upregulated genes",x="Timepoint", y = "log2 fold change SGC/DMSO") + theme_light()+
  theme(plot.title = element_text(size=15, face = "bold"), axis.title = element_text(size=10), axis.text = element_text(size=8, color = "black"), legend.position="none") + stat_compare_means(comparisons = my_list, label = "p.signif") +
  scale_color_manual(breaks = c("24 hours", "7 days"), values=c("grey", "#55A0FF"))

p2 <- ggplot(sum_down, aes(x=timepoint, y=lfc)) +
   geom_jitter(width=0.25, size = 0.05, aes(colour = timepoint)) +
  geom_boxplot(width=0.5, outlier.shape=NA, alpha = 0.2) +
  labs(title="SUM159 downregulated genes",x="Timepoint", y = "log2 fold change SGC/DMSO") + theme_light()+
  theme(plot.title = element_text(size=15, face = "bold"), axis.title = element_text(size=10), axis.text = element_text(size=8, color = "black"), legend.position="none") + stat_compare_means(comparisons = my_list, label = "p.signif") +
  scale_color_manual(breaks = c("24 hours", "7 days"), values=c("grey", "#55A0FF"))

p3 <- ggplot(mcf_up, aes(x=timepoint, y=lfc)) +
   geom_jitter(width=0.25, size = 0.05, aes(colour = timepoint)) +
  geom_boxplot(width=0.5, outlier.shape=NA, alpha = 0.2) +
  labs(title="MCF-7 upregulated genes",x="Timepoint", y = "log2 fold change SGC/DMSO") + theme_light()+
  theme(plot.title = element_text(size=15, face = "bold"), axis.title = element_text(size=10), axis.text = element_text(size=8, color = "black"), legend.position="none") + stat_compare_means(comparisons = my_list, label = "p.signif") +
  scale_color_manual(breaks = c("24 hours", "7 days"), values=c("grey", "#55A0FF"))

p4 <- ggplot(mcf_down, aes(x=timepoint, y=lfc)) +
   geom_jitter(width=0.25, size = 0.05, aes(colour = timepoint)) +
  geom_boxplot(width=0.5, outlier.shape=NA, alpha = 0.2) +
  labs(title="MCF-7 downregulated genes",x="Timepoint", y = "log2 fold change SGC/DMSO") + theme_light()+
  theme(plot.title = element_text(size=15, face = "bold"), axis.title = element_text(size=10), axis.text = element_text(size=8, color = "black"), legend.position="none") + stat_compare_means(comparisons = my_list, label = "p.signif") +
  scale_color_manual(breaks = c("24 hours", "7 days"), values=c("grey", "#55A0FF"))

grid.arrange(p1, p2, p3, p4, nrow=2)

```

