---
title: "SUM_Peaks"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(stringr)
library(org.Hs.eg.db)
library(clusterProfiler)
library(tidyverse)
```

## MLLT1 ChIP-Seq: SUM159

Previously - aligned .fastq files to .bam, removed duplicates. Used Homer to find peaks in DMSO samples, then annotate peaks using DMSO + SGC files. Analyse factor style peaks only. Used normalisation factor for SGC IP samples (calculated using Drosophila spike in to human DNA ratio)

```{r 7 days}
# read in files
all_data_7d_default <- read.delim("SUM_MLLT1_PEAKS_7D_FACTOR.quant.txt", header = TRUE, sep = "\t")
all_data_7d_norm <- read.delim("SUM_MLLT1_PEAKS_7D_FACTOR_NORM.quant.txt", header = TRUE, sep = "\t")

# move normalised SGC tag densities to default table (DMSO default normalisation value)
all_data_7d_default$SGC.Tag.Densities <- all_data_7d_norm$SUM_SGC_7D_IP..Tag.Count.in.given.bp..26479125.0.Total..normalization.factor...0.29..effective.total...7550759.462.

# get rid of default SGC tag density
all_data_7d <- subset(all_data_7d_default, select = -c(21))

# select specific columns and rearrange
peaks_7d <- all_data_7d[,c(16, 12, 20, 21, 17, 18, 19, 8, 9, 2, 3, 4)]

names(peaks_7d)[names(peaks_7d) == "SUM_DMSO_7D_IP..Tag.Count.in.given.bp..14966870.5.Total..normalization.factor...0.67..effective.total...10000000."] <- "DMSO.Tag.Densities"

# calculating fold change then log2 transform
peaks_7d$DMSO.Tag.Densities <- peaks_7d$DMSO.Tag.Densities + 1
peaks_7d$SGC.Tag.Densities <- peaks_7d$SGC.Tag.Densities + 1
peaks_7d$LFC <- peaks_7d$SGC.Tag.Densities/peaks_7d$DMSO.Tag.Densities
peaks_7d <- peaks_7d[,c(1, 2, 3, 4, 13, 5, 6, 7, 8, 9, 10, 11, 12)]
peaks_7d$LFC <- log2(peaks_7d$LFC)

# select only protein-coding genes
peaks_7d <-peaks_7d[!(peaks_7d$Gene.Type=="ncRNA"),]
peaks_7d <-peaks_7d[!(peaks_7d$Gene.Type=="pseudo"),]
write.csv(peaks_7d, "sum_peaks_7d.csv")
```

```{r 24 hours}
# read in files
all_data_24h_default <- read.delim("SUM_MLLT1_PEAKS_24H_FACTOR.quant.txt", header = TRUE, sep = "\t")
all_data_24h_norm <- read.delim("SUM_MLLT1_PEAKS_24H_FACTOR_NORM.quant.txt", header = TRUE, sep = "\t")

# move normalised SGC tag densities to default table (DMSO default normalisation value)
all_data_24h_default$SGC.Tag.Densities <- all_data_24h_norm$SUM_SGC_24H_IP..Tag.Count.in.given.bp..17595829.0.Total..normalization.factor...0.38..effective.total...6672871.029.

# get rid of default SGC tag density
all_data_24h <- subset(all_data_24h_default, select = -c(21))

# select specific columns and rearrange
peaks_24h <- all_data_24h[,c(16, 12, 20, 21, 17, 18, 19, 8, 9, 2, 3, 4)]

names(peaks_24h)[names(peaks_24h) == "SUM_DMSO_24H_IP..Tag.Count.in.given.bp..15957296.5.Total..normalization.factor...0.63..effective.total...10000000."] <- "DMSO.Tag.Densities"

# calculating fold change then log2 transform
peaks_24h$DMSO.Tag.Densities <- peaks_24h$DMSO.Tag.Densities + 1
peaks_24h$SGC.Tag.Densities <- peaks_24h$SGC.Tag.Densities + 1
peaks_24h$LFC <- peaks_24h$SGC.Tag.Densities/peaks_24h$DMSO.Tag.Densities
peaks_24h <- peaks_24h[,c(1, 2, 3, 4, 13, 5, 6, 7, 8, 9, 10, 11, 12)]
peaks_24h$LFC <- log2(peaks_24h$LFC)

# select only protein-coding genes
peaks_24h <-peaks_24h[!(peaks_24h$Gene.Type=="ncRNA"),]
peaks_24h <-peaks_24h[!(peaks_24h$Gene.Type=="pseudo"),]
write.csv(peaks_24h, "sum_peaks_24h.csv")
```

```{r GO}
# 7 days

# Keep only rows from table without NAs
peaks_7d_go <- peaks_7d[is.na(peaks_7d$Entrez.ID)==FALSE,]

# Remove duplicated entries
peaks_7d_go <- peaks_7d_go[!duplicated(peaks_7d_go$Entrez.ID),]

# select only peaks that decrease upon addition of inhibitor
peaks_7d_decrease <- peaks_7d_go[(peaks_7d_go$LFC < 0),]

# Change Entrez IDs from numbers to characters
geneset <- as.character(peaks_7d_go$Entrez.ID)

# This will take a little while to run
ego <- enrichGO(gene = geneset, 
                universe = NULL, #all available genes in database
                OrgDb = org.Hs.eg.db, #Hs: homo sapiens
                ont ="BP", #molecular function, biological process, cellular component
                pAdjustMethod = "BH",
                pvalueCutoff = 0.01,
                qvalueCutoff = 0.05,  #q value is FDR adjusted p value
                readable = TRUE) #will show gene symbol in images later rather than Entrez Gene ID
# dimensions - number of GO terms
dim(ego)

simp_ego <- simplify(ego)

dim(simp_ego)

# copy and paste table into Excel
sum_7d_GOgenes <- data.frame(simp_ego$ID, simp_ego$Description, simp_ego$p.adjust, simp_ego$geneID)
```


```{r GO}
# 24 hours

# Keep only rows from table without NAs
peaks_24h_go <- peaks_24h[is.na(peaks_24h$Entrez.ID)==FALSE,]

# Remove duplicated entries
peaks_24h_go <- peaks_24h_go[!duplicated(peaks_24h_go$Entrez.ID),]

# select only peaks that decrease upon addition of inhibitor
peaks_24h_decrease <- peaks_24h_go[(peaks_24h_go$LFC < 0),]

# Change Entrez IDs from numbers to characters
geneset <- as.character(peaks_24h_go$Entrez.ID)

# This will take a little while to run
ego <- enrichGO(gene = geneset, 
                universe = NULL, #all available genes in database
                OrgDb = org.Hs.eg.db, #Hs: homo sapiens
                ont ="BP", #molecular function, biological process, cellular component
                pAdjustMethod = "BH",
                pvalueCutoff = 0.01,
                qvalueCutoff = 0.05,  #q value is FDR adjusted p value
                readable = TRUE) #will show gene symbol in images later rather than Entrez Gene ID
# dimensions - number of GO terms
dim(ego)

simp_ego <- simplify(ego)

dim(simp_ego)

# copy and paste table into Excel
sum_24h_GOgenes <- data.frame(simp_ego$ID, simp_ego$Description, simp_ego$p.adjust, simp_ego$geneID)
```


```{r negative regulators}
# negative regulators of interferon response and antigen presentation & processing
neg_reg <- read.csv("negative_regulators.csv", header=TRUE)

# compare to each clone
peaks_neg_reg <- peaks_7d[is.element(peaks_7d$Gene.Name, neg_reg$Gene),]
write.csv(peaks_neg_reg, "sum_negative_regulators_peaks.csv")
```

```{r interleukin-1}
# interleukin-1 related genes
il1 <- read.csv("interleukin1.csv", header=TRUE)

# compare to each clone
peaks_il1 <- peaks_7d[is.element(peaks_7d$Gene.Name, il1$Gene),]
write.csv(peaks_il1, "sum_interleukin1_peaks.csv")

# pull out LFC values from RNA-Seq data

rnaseq_crispr1 <- read.csv("SUM1_res.csv", header=TRUE)
rnaseq_crispr2 <- read.csv("SUM2_res.csv", header=TRUE)
rnaseq_24h <- read.csv("SUM_DMSOvSGC_24H_res.csv", header=TRUE)
rnaseq_7d <- read.csv("SUM_DMSOvSGC_7D_res.csv", header=TRUE)

# SUM1 - pull out peaks from IL1 list, make gene names row names
rnaseq1 <- rnaseq_crispr1[is.element(rnaseq_crispr1$SYMBOL, peaks_il1$Gene.Name),]
rnaseq1 <- rnaseq1[c(1, 2)]
rownames(rnaseq1) <- rnaseq1$SYMBOL
rnaseq1[2] <- NULL

# merge table by rownames from SUM1 and 1st column in peaks table
peaks_lfc_il1 <- merge(rnaseq1, peaks_il1, by.x = 0, by.y = 1, all=TRUE)

# rename columns
names(peaks_lfc_il1)[names(peaks_lfc_il1) == "log2FoldChange"] <- "SUM1_LFC"

# SUM2 - pull out peaks from IL1 list, make gene names row names
rnaseq2 <- rnaseq_crispr2[is.element(rnaseq_crispr2$SYMBOL, peaks_il1$Gene.Name),]
rnaseq2 <- rnaseq2[c(1, 2)]

# merge table by rownames from SUM1 and 1st column in peaks table
peaks_lfc_il1 <- merge(rnaseq2, peaks_lfc_il1, by.x = 2, by.y = 1, all=TRUE)

# rename columns
names(peaks_lfc_il1)[names(peaks_lfc_il1) == "log2FoldChange"] <- "SUM2_LFC"

# SUM 24H - pull out peaks from IL1 list, make gene names row names
rnaseq3 <- rnaseq_24h[is.element(rnaseq_24h$SYMBOL, peaks_il1$Gene.Name),]
rnaseq3 <- rnaseq3[c(1, 2)]

# merge table by rownames from SUM1 and 1st column in peaks table
peaks_lfc_il1 <- merge(rnaseq3, peaks_lfc_il1, by.x = 2, by.y = 1, all=TRUE)

# rename columns
names(peaks_lfc_il1)[names(peaks_lfc_il1) == "log2FoldChange"] <- "SUM24H_LFC"

# SUM 7D - pull out peaks from IL1 list, make gene names row names
rnaseq4 <- rnaseq_7d[is.element(rnaseq_7d$SYMBOL, peaks_il1$Gene.Name),]
rnaseq4 <- rnaseq4[c(1, 2)]

# merge table by rownames from SUM1 and 1st column in peaks table
peaks_lfc_il1 <- merge(rnaseq4, peaks_lfc_il1, by.x = 2, by.y = 1, all=TRUE)

# rename columns
names(peaks_lfc_il1)[names(peaks_lfc_il1) == "log2FoldChange"] <- "SUM7D_LFC"

write.csv(peaks_lfc_il1, "sum_peaks_lfc_il1.csv")


```

```{r interferon alpha and interferon gamma}
# see if peak list for interleukin-1 response genes overlap with genes found in Hallmark gene sets interferon alpha and interferon gamma response (downregulated in SGC-iMLLT RNA-Seq)

# interferon alpha and interferon gamma core enriched genes
ifna_ifng <- read.csv("ifna_ifng.csv", header=TRUE)


ifn_response <- intersect(ifna_ifng$Genes, il1$Gene)

# only 8 genes in common PSMB8, PSMB10, NFKBIA, HIF1A, MAP3K8, HAS2, CXCL8, SRC

different <- setdiff(ifna_ifng$Genes, il1$Gene)


```
