---
title: "merged_v_crispr"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(limma)
library(reshape)
library(reshape2)
library("RColorBrewer")
library(wesanderson)

```

# CRISPR RNA-Seq and Merged SGC-iMLLT RNA-Seq

Venn diagrams to show common and non-common up/downregulated genes
Gene ontology (biological processes) - barplots only

```{r mcf genes}
# read in gene lists (total differentially expressed genes)
crispr_1 <- read.csv("results/MCF/CRISPR_MCF1_res.csv", header=TRUE)
crispr_2 <- read.csv("results/MCF/CRISPR_MCF2_res.csv", header=TRUE)
sgc_24h <- read.csv("results/MCF/Merge_MCF_DMSOvSGC_24H_res.csv", header=TRUE)
sgc_7d <- read.csv("results/MCF/Merge_MCF_DMSOvSGC_7D_res.csv", header=TRUE)
```

## MCF-7

### CRISPR + SGC-iMLLT 24 hours

#### Downregulated

```{r}
# filter lfc < 0 for downregulated
sgc_24h <- sgc_24h %>%
  filter(log2FoldChange < 0)
crispr_1 <- crispr_1 %>% 
  filter(log2FoldChange < 0)
crispr_2 <- crispr_2 %>% 
  filter(log2FoldChange < 0)

# select symbol and lfc columns from 24 hour tables
sgc_24h <- sgc_24h %>% 
  select(SYMBOL, log2FoldChange)
crispr_1 <- crispr_1 %>% 
  select(SYMBOL, log2FoldChange)
crispr_2 <- crispr_2 %>% 
  select(SYMBOL, log2FoldChange)

# combine them by symbol column
combine <- merge(sgc_24h, crispr_1, by="SYMBOL", all = TRUE)

# rename two lfc columns with specific names
combine <- combine %>% 
  mutate(sgc_24h_LFC = log2FoldChange.x, crispr_1_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

# add crispr 2 column
combine <- merge(combine, crispr_2, by="SYMBOL", all = TRUE)
combine <- combine %>% 
  mutate(crispr_2_LFC = log2FoldChange) %>% 
  select(-log2FoldChange)

# remove any rows that have NA
combine[is.na(combine)] <- 0

# select just lfc columns for Venn diagram
sgc_24h_venn <- combine$sgc_24h_LFC
crispr_1_venn <- combine$crispr_1_LFC
crispr_2_venn <- combine$crispr_2_LFC

# combine into single object and generate Venn diagram
c2 <- cbind(sgc_24h_venn, crispr_1_venn, crispr_2_venn)
a <- vennCounts(c2)
a
c("red", "blue", "darkgreen", "yellow")

png("venn2/mcf_CRISPR_SGC_24h_shared_genes_down.png")

vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

combine_24h <- combine[complete.cases(combine),]

write.csv(combine_24h, file="venn2/mcf_CRISPR_SGC_24h_shared_genes_down.csv")

# want only genes that are shared in both runs and CRISPR

# substitute NA for 0, then remove any rows with NA
combine_24h[combine_24h==0] <- NA
combine_24h<-combine_24h[complete.cases(combine_24h),]
write.csv(combine_24h, file="go2/mcf_shared_genes_24h_down.csv")
```
![](venn2/mcf_CRISPR_SGC_24h_shared_genes_down.png)
#### Upregulated

```{r}
# filter lfc > 0 for upregulated
sgc_24h <- sgc_24h %>%
  filter(log2FoldChange > 0)
crispr_1 <- crispr_1 %>% 
  filter(log2FoldChange > 0)
crispr_2 <- crispr_2 %>% 
  filter(log2FoldChange > 0)

# select symbol and lfc columns from 24 hour tables
sgc_24h <- sgc_24h %>% 
  select(SYMBOL, log2FoldChange)
crispr_1 <- crispr_1 %>% 
  select(SYMBOL, log2FoldChange)
crispr_2 <- crispr_2 %>% 
  select(SYMBOL, log2FoldChange)

# combine them by symbol column
combine <- merge(sgc_24h, crispr_1, by="SYMBOL", all = TRUE)

# rename two lfc columns with specific names
combine <- combine %>% 
  mutate(sgc_24h_LFC = log2FoldChange.x, crispr_1_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

# add crispr 2 column
combine <- merge(combine, crispr_2, by="SYMBOL", all = TRUE)
combine <- combine %>% 
  mutate(crispr_2_LFC = log2FoldChange) %>% 
  select(-log2FoldChange)

# remove any rows that have NA
combine[is.na(combine)] <- 0

# select just lfc columns for Venn diagram
sgc_24h_venn <- combine$sgc_24h_LFC
crispr_1_venn <- combine$crispr_1_LFC
crispr_2_venn <- combine$crispr_2_LFC

# combine into single object and generate Venn diagram
c2 <- cbind(sgc_24h_venn, crispr_1_venn, crispr_2_venn)
a <- vennCounts(c2)
a
c("red", "blue", "darkgreen", "yellow")

png("venn2/mcf_CRISPR_SGC_24h_shared_genes_up.png")

vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

combine_24h <- combine[complete.cases(combine),]

write.csv(combine_24h, file="venn2/mcf_CRISPR_SGC_24h_shared_genes_up.csv")

# want only genes that are shared in both runs and CRISPR

# substitute NA for 0, then remove any rows with NA
combine_24h[combine_24h==0] <- NA
combine_24h<-combine_24h[complete.cases(combine_24h),]
write.csv(combine_24h, file="go2/mcf_shared_genes_24h_up.csv")
```
![](venn2/mcf_CRISPR_SGC_24h_shared_genes_up.png)



### CRISPR + SGC-iMLLT 7 days

#### Downregulated

```{r}
# filter lfc < 0 for downregulated
sgc_7d <- sgc_7d %>%
  filter(log2FoldChange < 0)
crispr_1 <- crispr_1 %>% 
  filter(log2FoldChange < 0)
crispr_2 <- crispr_2 %>% 
  filter(log2FoldChange < 0)

# select symbol and lfc columns from 24 hour tables
sgc_7d <- sgc_7d %>% 
  select(SYMBOL, log2FoldChange)
crispr_1 <- crispr_1 %>% 
  select(SYMBOL, log2FoldChange)
crispr_2 <- crispr_2 %>% 
  select(SYMBOL, log2FoldChange)

# combine them by symbol column
combine <- merge(sgc_7d, crispr_1, by="SYMBOL", all = TRUE)

# rename two lfc columns with specific names
combine <- combine %>% 
  mutate(sgc_7d_LFC = log2FoldChange.x, crispr_1_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

# add crispr 2 column
combine <- merge(combine, crispr_2, by="SYMBOL", all = TRUE)
combine <- combine %>% 
  mutate(crispr_2_LFC = log2FoldChange) %>% 
  select(-log2FoldChange)

# remove any rows that have NA
combine[is.na(combine)] <- 0

# select just lfc columns for Venn diagram
sgc_7d_venn <- combine$sgc_7d_LFC
crispr_1_venn <- combine$crispr_1_LFC
crispr_2_venn <- combine$crispr_2_LFC

# combine into single object and generate Venn diagram
c2 <- cbind(sgc_7d_venn, crispr_1_venn, crispr_2_venn)
a <- vennCounts(c2)
a
c("red", "blue", "darkgreen", "yellow")

png("venn2/mcf_CRISPR_SGC_7d_shared_genes_down.png")

vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

combine_7d <- combine[complete.cases(combine),]

write.csv(combine_7d, file="venn2/mcf_CRISPR_SGC_7d_shared_genes_down.csv")

# want only genes that are shared in both runs and CRISPR

# substitute NA for 0, then remove any rows with NA
combine_7d[combine_7d==0] <- NA
combine_7d<-combine_7d[complete.cases(combine_7d),]
write.csv(combine_7d, file="go2/mcf_shared_genes_7d_down.csv")
```
![](venn2/mcf_CRISPR_SGC_7d_shared_genes_down.png)
#### Upregulated

```{r}
# filter lfc > 0 for upregulated
sgc_7d <- sgc_7d %>%
  filter(log2FoldChange > 0)
crispr_1 <- crispr_1 %>% 
  filter(log2FoldChange > 0)
crispr_2 <- crispr_2 %>% 
  filter(log2FoldChange > 0)

# select symbol and lfc columns from 24 hour tables
sgc_7d <- sgc_7d %>% 
  select(SYMBOL, log2FoldChange)
crispr_1 <- crispr_1 %>% 
  select(SYMBOL, log2FoldChange)
crispr_2 <- crispr_2 %>% 
  select(SYMBOL, log2FoldChange)

# combine them by symbol column
combine <- merge(sgc_7d, crispr_1, by="SYMBOL", all = TRUE)

# rename two lfc columns with specific names
combine <- combine %>% 
  mutate(sgc_7d_LFC = log2FoldChange.x, crispr_1_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

# add crispr 2 column
combine <- merge(combine, crispr_2, by="SYMBOL", all = TRUE)
combine <- combine %>% 
  mutate(crispr_2_LFC = log2FoldChange) %>% 
  select(-log2FoldChange)

# remove any rows that have NA
combine[is.na(combine)] <- 0

# select just lfc columns for Venn diagram
sgc_7d_venn <- combine$sgc_7d_LFC
crispr_1_venn <- combine$crispr_1_LFC
crispr_2_venn <- combine$crispr_2_LFC

# combine into single object and generate Venn diagram
c2 <- cbind(sgc_7d_venn, crispr_1_venn, crispr_2_venn)
a <- vennCounts(c2)
a
c("red", "blue", "darkgreen", "yellow")

png("venn2/mcf_CRISPR_SGC_7d_shared_genes_up.png")

vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

combine_7d <- combine[complete.cases(combine),]

write.csv(combine_7d, file="venn2/mcf_CRISPR_SGC_7d_shared_genes_up.csv")

# want only genes that are shared in both runs and CRISPR

# substitute NA for 0, then remove any rows with NA
combine_7d[combine_7d==0] <- NA
combine_7d<-combine_7d[complete.cases(combine_7d),]
write.csv(combine_7d, file="go2/mcf_shared_genes_7d_up.csv")
```
![](venn2/mcf_CRISPR_SGC_7d_shared_genes_up.png)


```{r sum genes}
# read in gene lists (total differentially expressed genes)
crispr_1 <- read.csv("results/SUM/CRISPR_SUM1_res.csv", header=TRUE)
crispr_2 <- read.csv("results/SUM/CRISPR_SUM2_res.csv", header=TRUE)
sgc_24h <- read.csv("results/SUM/Merge_SUM_DMSOvSGC_24H_res.csv", header=TRUE)
sgc_7d <- read.csv("results/SUM/Merge_SUM_DMSOvSGC_7D_res.csv", header=TRUE)
```

## SUM159

### CRISPR + SGC-iMLLT 24 hours

#### Downregulated

```{r}
# filter lfc < 0 for downregulated
sgc_24h <- sgc_24h %>%
  filter(log2FoldChange < 0)
crispr_1 <- crispr_1 %>% 
  filter(log2FoldChange < 0)
crispr_2 <- crispr_2 %>% 
  filter(log2FoldChange < 0)

# select symbol and lfc columns from 24 hour tables
sgc_24h <- sgc_24h %>% 
  select(SYMBOL, log2FoldChange)
crispr_1 <- crispr_1 %>% 
  select(SYMBOL, log2FoldChange)
crispr_2 <- crispr_2 %>% 
  select(SYMBOL, log2FoldChange)

# combine them by symbol column
combine <- merge(sgc_24h, crispr_1, by="SYMBOL", all = TRUE)

# rename two lfc columns with specific names
combine <- combine %>% 
  mutate(sgc_24h_LFC = log2FoldChange.x, crispr_1_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

# add crispr 2 column
combine <- merge(combine, crispr_2, by="SYMBOL", all = TRUE)
combine <- combine %>% 
  mutate(crispr_2_LFC = log2FoldChange) %>% 
  select(-log2FoldChange)

# remove any rows that have NA
combine[is.na(combine)] <- 0

# select just lfc columns for Venn diagram
sgc_24h_venn <- combine$sgc_24h_LFC
crispr_1_venn <- combine$crispr_1_LFC
crispr_2_venn <- combine$crispr_2_LFC

# combine into single object and generate Venn diagram
c2 <- cbind(sgc_24h_venn, crispr_1_venn, crispr_2_venn)
a <- vennCounts(c2)
a
c("red", "blue", "darkgreen", "yellow")

png("venn2/sum_CRISPR_SGC_24h_shared_genes_down.png")

vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

combine_24h <- combine[complete.cases(combine),]

write.csv(combine_24h, file="venn2/sum_CRISPR_SGC_24h_shared_genes_down.csv")

# want only genes that are shared in both runs and CRISPR

# substitute NA for 0, then remove any rows with NA
combine_24h[combine_24h==0] <- NA
combine_24h<-combine_24h[complete.cases(combine_24h),]
write.csv(combine_24h, file="go2/sum_shared_genes_24h_down.csv")
```
![](venn2/sum_CRISPR_SGC_24h_shared_genes_down.png)
#### Upregulated

```{r}
# filter lfc > 0 for upregulated
sgc_24h <- sgc_24h %>%
  filter(log2FoldChange > 0)
crispr_1 <- crispr_1 %>% 
  filter(log2FoldChange > 0)
crispr_2 <- crispr_2 %>% 
  filter(log2FoldChange > 0)

# select symbol and lfc columns from 24 hour tables
sgc_24h <- sgc_24h %>% 
  select(SYMBOL, log2FoldChange)
crispr_1 <- crispr_1 %>% 
  select(SYMBOL, log2FoldChange)
crispr_2 <- crispr_2 %>% 
  select(SYMBOL, log2FoldChange)

# combine them by symbol column
combine <- merge(sgc_24h, crispr_1, by="SYMBOL", all = TRUE)

# rename two lfc columns with specific names
combine <- combine %>% 
  mutate(sgc_24h_LFC = log2FoldChange.x, crispr_1_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

# add crispr 2 column
combine <- merge(combine, crispr_2, by="SYMBOL", all = TRUE)
combine <- combine %>% 
  mutate(crispr_2_LFC = log2FoldChange) %>% 
  select(-log2FoldChange)

# remove any rows that have NA
combine[is.na(combine)] <- 0

# select just lfc columns for Venn diagram
sgc_24h_venn <- combine$sgc_24h_LFC
crispr_1_venn <- combine$crispr_1_LFC
crispr_2_venn <- combine$crispr_2_LFC

# combine into single object and generate Venn diagram
c2 <- cbind(sgc_24h_venn, crispr_1_venn, crispr_2_venn)
a <- vennCounts(c2)
a
c("red", "blue", "darkgreen", "yellow")

png("venn2/sum_CRISPR_SGC_24h_shared_genes_up.png")

vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

combine_24h <- combine[complete.cases(combine),]

write.csv(combine_24h, file="venn2/sum_CRISPR_SGC_24h_shared_genes_up.csv")

# want only genes that are shared in both runs and CRISPR

# substitute NA for 0, then remove any rows with NA
combine_24h[combine_24h==0] <- NA
combine_24h<-combine_24h[complete.cases(combine_24h),]
write.csv(combine_24h, file="go2/sum_shared_genes_24h_up.csv")
```
![](venn2/sum_CRISPR_SGC_24h_shared_genes_up.png)



### CRISPR + SGC-iMLLT 7 days

#### Downregulated

```{r}
# filter lfc < 0 for downregulated
sgc_7d <- sgc_7d %>%
  filter(log2FoldChange < 0)
crispr_1 <- crispr_1 %>% 
  filter(log2FoldChange < 0)
crispr_2 <- crispr_2 %>% 
  filter(log2FoldChange < 0)

# select symbol and lfc columns from 24 hour tables
sgc_7d <- sgc_7d %>% 
  select(SYMBOL, log2FoldChange)
crispr_1 <- crispr_1 %>% 
  select(SYMBOL, log2FoldChange)
crispr_2 <- crispr_2 %>% 
  select(SYMBOL, log2FoldChange)

# combine them by symbol column
combine <- merge(sgc_7d, crispr_1, by="SYMBOL", all = TRUE)

# rename two lfc columns with specific names
combine <- combine %>% 
  mutate(sgc_7d_LFC = log2FoldChange.x, crispr_1_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

# add crispr 2 column
combine <- merge(combine, crispr_2, by="SYMBOL", all = TRUE)
combine <- combine %>% 
  mutate(crispr_2_LFC = log2FoldChange) %>% 
  select(-log2FoldChange)

# remove any rows that have NA
combine[is.na(combine)] <- 0

# select just lfc columns for Venn diagram
sgc_7d_venn <- combine$sgc_7d_LFC
crispr_1_venn <- combine$crispr_1_LFC
crispr_2_venn <- combine$crispr_2_LFC

# combine into single object and generate Venn diagram
c2 <- cbind(sgc_7d_venn, crispr_1_venn, crispr_2_venn)
a <- vennCounts(c2)
a
c("red", "blue", "darkgreen", "yellow")

png("venn2/sum_CRISPR_SGC_7d_shared_genes_down.png")

vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

combine_7d <- combine[complete.cases(combine),]

write.csv(combine_7d, file="venn2/sum_CRISPR_SGC_7d_shared_genes_down.csv")

# want only genes that are shared in both runs and CRISPR

# substitute NA for 0, then remove any rows with NA
combine_7d[combine_7d==0] <- NA
combine_7d<-combine_7d[complete.cases(combine_7d),]
write.csv(combine_7d, file="go2/sum_shared_genes_7d_down.csv")
```
![](venn2/sum_CRISPR_SGC_7d_shared_genes_down.png)
#### Upregulated

```{r}
# filter lfc > 0 for upregulated
sgc_7d <- sgc_7d %>%
  filter(log2FoldChange > 0)
crispr_1 <- crispr_1 %>% 
  filter(log2FoldChange > 0)
crispr_2 <- crispr_2 %>% 
  filter(log2FoldChange > 0)

# select symbol and lfc columns from 24 hour tables
sgc_7d <- sgc_7d %>% 
  select(SYMBOL, log2FoldChange)
crispr_1 <- crispr_1 %>% 
  select(SYMBOL, log2FoldChange)
crispr_2 <- crispr_2 %>% 
  select(SYMBOL, log2FoldChange)

# combine them by symbol column
combine <- merge(sgc_7d, crispr_1, by="SYMBOL", all = TRUE)

# rename two lfc columns with specific names
combine <- combine %>% 
  mutate(sgc_7d_LFC = log2FoldChange.x, crispr_1_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

# add crispr 2 column
combine <- merge(combine, crispr_2, by="SYMBOL", all = TRUE)
combine <- combine %>% 
  mutate(crispr_2_LFC = log2FoldChange) %>% 
  select(-log2FoldChange)

# remove any rows that have NA
combine[is.na(combine)] <- 0

# select just lfc columns for Venn diagram
sgc_7d_venn <- combine$sgc_7d_LFC
crispr_1_venn <- combine$crispr_1_LFC
crispr_2_venn <- combine$crispr_2_LFC

# combine into single object and generate Venn diagram
c2 <- cbind(sgc_7d_venn, crispr_1_venn, crispr_2_venn)
a <- vennCounts(c2)
a
c("red", "blue", "darkgreen", "yellow")

png("venn2/sum_CRISPR_SGC_7d_shared_genes_up.png")

vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

combine_7d <- combine[complete.cases(combine),]

write.csv(combine_7d, file="venn2/sum_CRISPR_SGC_7d_shared_genes_up.csv")

# want only genes that are shared in both runs and CRISPR

# substitute NA for 0, then remove any rows with NA
combine_7d[combine_7d==0] <- NA
combine_7d<-combine_7d[complete.cases(combine_7d),]
write.csv(combine_7d, file="go2/sum_shared_genes_7d_up.csv")
```
![](venn2/sum_CRISPR_SGC_7d_shared_genes_up.png)
