---
title: "Differential expression using DESeq2"
output:
  html_document: default
  always_allow_html: true
  word_document: default
---

Update: 8th July 2020

# Differential gene expression using DESeq2 - MCF-7

Differential gene expression using DESeq2 and visualisation with Venn diagrams. Using Adam's code, some from https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html


```{r, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
source('functions.R')
library(org.Hs.eg.db)
library(AnnotationDbi)
library(DESeq2)
library(pheatmap)
library(tidyverse)
library(DT)
library(limma)
library(pheatmap)

# to convert to HTML document, type in Console > rmarkdown::render(input = "file_name.Rmd", output_format = "html_document", output_file = NULL) - have to do HTML due to tab layout

```


```{r}

meta_data <- read.table("mcf_metadata.csv", sep=",", header = TRUE)

rownames(meta_data) <- meta_data$Samples

colnames(txi.mcf_kallisto$counts) <- rownames(meta_data)

all(rownames(meta_data) %in% colnames(txi.mcf_kallisto$counts))

```


```{r}
# set up meta_data
# meta_data_1 <- meta_data %>% 
  # filter(Cell_line == "MCF") %>% 
  # column_to_rownames("Samples")

# giving an error saying .data must be a data frame without row names. Don't need to use, as set up separate metadata sheets for SUM159 and MCF-7 anyway

```




```{r, message=FALSE, eval=FALSE}
## Set up the DESeq2 object

df_mRNA <-  txi.mcf_kallisto

deseq2_txi <- function(df_mRNA, meta_data, control="dmso", test="sgc", value){

    dds<- DESeqDataSetFromTximport(df_mRNA, meta_data, ~Condition)
  
  keep <- rowSums(counts(dds)) >= 10
  dds <- dds[keep,]
  
  dds <- DESeq(dds)
  
  res <- results(dds, contrast = c(value, test, control))
  
  return(res)
}

res_24H <- deseq2_txi(df_mRNA, meta_data, control = "DMSO_24H", test="SGC_24H", value="Condition")

res_7D <- deseq2_txi(df_mRNA, meta_data, control = "DMSO_7D", test="SGC_7D", value="Condition")


```

The following describes the analysis of the data using DEseq2. I have separated the analyses into three different contrasts and have plotted summary statistics and MA plots for each.  


## Summary of the data

### DMSO vs SGC at 24 hours

```{r}
summary(res_24H, alpha=0.05)
```

### DMSO vs SGC at 7 days

```{r}
summary(res_7D, alpha=0.05)
```



## MA plots 

In DESeq2, the function plotMA shows the log2 fold changes attributable to a given variable over the mean of normalized counts for all the samples in the DESeqDataSet. Points will be colored blue if the adjusted p value is less than 0.01. Points which fall out of the window are plotted as open triangles pointing either up or down.

### DMSO vs SGC at 24 hours

```{r, fig.width=8, fig.height=5}
DESeq2::plotMA(res_24H)
```

### DMSO vs SGC at 7 days

```{r, fig.width=8, fig.height=5}
DESeq2::plotMA(res_7D)
```



## Volcano plots 

### DMSO vs SGC at 24 hours

```{r, message=FALSE, , fig.width=6, fig.height=6}

library("ggplot2") #Best plots
library("ggrepel") #Avoid overlapping labels

test <- as.data.frame(res_24H)
  
data <- as.vector(rownames(test))
annots <-  AnnotationDbi::select(org.Hs.eg.db, keys=data,
                                   columns="SYMBOL", keytype = "ENSEMBLTRANS")
  
result <- merge(test, annots, by.x="row.names", by.y="ENSEMBLTRANS")
res <- result %>% 
    dplyr::select(log2FoldChange, SYMBOL, baseMean, padj, Row.names) %>% 
    na.omit()
  
mutateddf <- mutate(res, sig=ifelse(res$padj<0.01, "padj<0.01", "Not Sig")) #Will have different colors depending on significance
input <- cbind(gene=rownames(mutateddf), mutateddf )
input <- input %>% 
  arrange(input$padj)

#convert the rownames to a column
volc = ggplot(input, aes(log2FoldChange, -log10(padj))) + #volcanoplot with log2Foldchange versus pvalue
    geom_point(aes(col=sig)) + #add points colored by significance
    ggtitle("DMSO v SGC at 24 hours") #e.g. 'Volcanoplot DESeq2'
volc

#Save Volcano Plot
# ggsave("volcano/MCF_DMSOvSGC_24H.jpeg", device="jpeg") #In case you want to easily save to disk

```

### DMSO vs SGC at 7 days

```{r, message=FALSE, fig.width=6, fig.height=6}

library("ggplot2") #Best plots
library("ggrepel") #Avoid overlapping labels

test <- as.data.frame(res_7D)
  
data <- as.vector(rownames(test))
annots <-  AnnotationDbi::select(org.Hs.eg.db, keys=data,
                                   columns="SYMBOL", keytype = "ENSEMBLTRANS")
  
result <- merge(test, annots, by.x="row.names", by.y="ENSEMBLTRANS")
res <- result %>% 
    dplyr::select(log2FoldChange, SYMBOL, baseMean, padj, Row.names) %>% 
    na.omit()
  
mutateddf <- mutate(res, sig=ifelse(res$padj<0.01, "padj<0.01", "Not Sig")) #Will have different colors depending on significance
input <- cbind(gene=rownames(mutateddf), mutateddf )
input <- input %>% 
  arrange(input$padj)

#convert the rownames to a column
volc = ggplot(input, aes(log2FoldChange, -log10(padj))) + #volcanoplot with log2Foldchange versus pvalue
    geom_point(aes(col=sig)) + #add points colored by significance
    ggtitle("DMSO v SGC at 7 days") #e.g. 'Volcanoplot DESeq2'
volc

#Save Volcano Plot
# ggsave("volcano/MCF_DMSOvSGC_7D.jpeg", device="jpeg") #In case you want to easily save to disk

```





## Results tables

The folowing results tables show the significant genes (p<0.05). Filtering has been performed with a log2 fold change +/- 2. These have been exported as .csv files where _res.csv is all genes, and _sig.csv is significant genes (p<0.05) with LFC +/-2 (see results folder)


### DMSO vs SGC at 24 hours

```{r, message=FALSE}

# altered filter_genes function to work with Ensembl Transcript IDs
dt <- filter_genes(as.data.frame(res_24H), name="MCF_DMSOvSGC_24H")

datatable(dt$sig)

```


### DMSO vs SGC at 7 days

```{r, message=FALSE}
dt <- filter_genes(as.data.frame(res_7D), name="MCF_DMSOvSGC_7D")

datatable(dt$sig)

```




## Genes common between 24h and 7d treatments
Too few genes to sort into up/downregulated. Selected only significant genes from results table p<0.05 (not taking into account value of log fold change).

```{r}

res_24h <- read.csv("results/MCF_DMSOvSGC_24H_res.csv") 
sig_24h <- res_24h[res_24h$padj<0.05 ,]

res_7d <- read.csv("results/MCF_DMSOvSGC_7D_res.csv") 
sig_7d <- res_7d[res_7d$padj<0.05 ,]

common_genes <- sig_7d[is.element(sig_7d$SYMBOL, sig_24h$SYMBOL),]

datatable(common_genes)

```

## Venn diagrams to show shared genes between DMSOvSGC 24 hours and DMSOvSGC 7 days

```{r}
# downregulated

res_24h_down <- res_24h %>% 
  filter(log2FoldChange < 0) %>% 
  filter(padj < 0.01)

res_7d_down <- res_7d %>% 
  filter(log2FoldChange < 0) %>% 
  filter(padj < 0.01)


# upregulated

res_24h_up <- res_24h %>% 
  filter(log2FoldChange > 0)%>% 
  filter(padj < 0.01)

res_7d_up <- res_7d %>% 
  filter(log2FoldChange > 0)%>% 
  filter(padj < 0.01)

```

### Downregulated genes

```{r, message=FALSE, eval=FALSE}

library(reshape)
library(reshape2)

res_24h_down <- res_24h_down %>% 
  select(SYMBOL, log2FoldChange)
res_7d_down <- res_7d_down %>% 
  select(SYMBOL, log2FoldChange)

merge_down <- merge(res_24h_down, res_7d_down, by="SYMBOL", all = TRUE)

merge_down <- merge_down %>% 
  mutate(res_24h_LFC = log2FoldChange.x, res_7d_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

merge_down[is.na(merge_down)] <- 0

res_24h_venn <- (merge_down$res_24h_LFC < 0)
res_7d_venn <- (merge_down$res_7d_LFC < 0)

c2 <- cbind(res_24h_venn, res_7d_venn)

a <- vennCounts(c2)
a

library("RColorBrewer")
library(wesanderson)
c("red", "blue", "darkgreen", "yellow")
png("venn/MCF_downregulated_venn.png")
vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

down <- merge_down[complete.cases(merge_down),]

write.csv(down, file="venn/MCF_down.csv")

```

![](venn/MCF_downregulated_venn.png)

### Upregulated genes

```{r, message=FALSE, eval=FALSE}

library(reshape)
library(reshape2)

res_24h_up <- res_24h_up %>% 
  select(SYMBOL, log2FoldChange)
res_7d_up <- res_7d_up %>% 
  select(SYMBOL, log2FoldChange)

merge_up <- merge(res_24h_up, res_7d_up, by="SYMBOL", all = TRUE)

merge_up <- merge_up %>% 
  mutate(res_24h_LFC = log2FoldChange.x, res_7d_LFC = log2FoldChange.y) %>% 
  select(-log2FoldChange.x, -log2FoldChange.y)

merge_up[is.na(merge_up)] <- 0

res_24h_venn <- (merge_up$res_24h_LFC < 0)
res_7d_venn <- (merge_up$res_7d_LFC < 0)

c2 <- cbind(res_24h_venn, res_7d_venn)

a <- vennCounts(c2)
a

library("RColorBrewer")
library(wesanderson)
c("red", "blue", "darkgreen", "yellow")
png("venn/MCF_upregulated_venn.png")
vennDiagram(a, circle.col = wes_palette("Darjeeling1"))

up <- merge_up[complete.cases(merge_up),]

write.csv(up, file="venn/MCF_up.csv")

```
![](venn/MCF_upregulated_venn.png)
