---
title: "imsig"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(imsig)
library(dplyr)
library(tidyverse)
library(gplots)
library(RColorBrewer)
library(pheatmap)
library(DESeq2)
library(ggplot2)
library (stringi)
library(AnnotationDbi)
library(edgeR)
library(DEFormats)
library(statmod)
library(BiocParallel)
library (limma)
library("genefilter")
```

Update 7th October 2020

# ImgSig package to interrogate METABRIC dataset

Immune cell signature signatures for profiling the microenvironment of solid tumours https://cancerimmunolres.aacrjournals.org/content/6/11/1388

The genomic and transcriptomic architecture of 2,000 breast tumours reveals novel subgroups (METABRIC) https://www.nature.com/articles/nature10983

Should be able to download METABRIC dataset from European Genome-Phenome Archive (http://www.ebi.ac.uk/ega/) EGAS00000000083, but site seems to be down for now.

Managed to download from cBioPortal

Keep separate from general METABRIC dataset analysis

```{r read in files}
clinical_patient <- read.delim("data_clinical_patient.txt", header=TRUE, na.strings = c("", "NA"))

expression_median <- read.delim("data_expression_median.txt", header=TRUE, na.strings = c("", "NA"))

# read in patient IDs

high_ids <- read.csv("high_patient_ids.csv", header=TRUE)

low_ids <- read.csv("low_patient_ids.csv", header=TRUE)

# convert to character by removing duplicates, then remove brca_metabric from each name

high_ids <-  high_ids[!duplicated(high_ids), ]

high_ids <- high_ids %>% str_replace("brca_metabric:", "")

low_ids <-  low_ids[!duplicated(low_ids), ]

low_ids <- low_ids %>% str_replace("brca_metabric:", "")

```

```{r preparing files for imsig}
# preparing expression data for ImSig
exp <- expression_median

# remove EntrezID
exp[2] <- NULL

# change . in patient IDs to - to match clinical data patient IDs
names(exp) <- gsub(x = names(exp), pattern = "\\.", replacement = "-")  

# remove any NA
exp <-  exp %>% na.omit()

# make gene symbols row names, and remove 1st column
rownames(exp) <- exp[,1]
exp[1] <- NULL

# high MLLT1 patients
high_exp <- exp[high_ids]

# low MLLT1 patients
low_exp <- exp[low_ids]

# preparing clinical data - keep only patient ID, overall survival in months and status (live/dead)
cli <- clinical_patient[,c(1, 13, 14)]

# remove top 4 rows (descriptive)
cli <- cli[-c(1,2,3, 4),]

# change column headings
names(cli)[names(cli) == "Overall.Survival..Months."] <- "time"
names(cli)[names(cli) == "Overall.Survival.Status"] <- "status"

# make patient IDs row names and remove column
rownames(cli) <- cli[,1]
cli[1] <- NULL

# remove row with no survival data (NA)
cli <-  cli %>% na.omit()

# change status to 0 and 1
cli$status[cli$status =="0:LIVING"] <- 0
cli$status[cli$status =="1:DECEASED"] <- 1
```

```{r running ImSig functions}

# table of stats on data
# high_stats <-  gene_stat(exp = high_exp, r = 0.7)

# run imsig algorithm
# cell_abundance <- imsig (exp = high_exp, r=0.6)


# Mean Expression Function

imsig <- function(exp,sig){
  # Subset genes that are present in the sig
  exp <- exp[row.names(exp) %in% sig$gene,]
  sig <- sig[sig$gene %in% row.names(exp),]
  # Loop to calculate the average expression of each cell type
  cc <- data.frame(matrix(nrow = ncol(exp)))
  cc <- cc[,-1]
  for (i in unique(sig$cell)){
    s <- sig[sig$cell %in% i,]
    e <- exp[as.character(s$gene),]
    e_avg <- data.frame(colMeans(e, na.rm = TRUE))
    colnames(e_avg) <- i
    cc <- cbind(cc, e_avg)
  }
  return(cc)
}

# Plotting Function

plot_abundance <- function(proportion){
  require(ggplot2)
  require(gridExtra)
  cell <- proportion
  cell$samples <- row.names(cell)
  cell$samples <- factor(cell$samples, levels = cell$samples)
  plots = lapply(1:(ncol(cell)-1), function(x) ggplot(cell, aes(x = cell$samples, y = cell[,x]))
                 + geom_bar(stat = "identity") + theme_classic() +
                   theme(axis.title.x=element_blank(), axis.text.x = element_text(angle = 90, hjust = 1), axis.title.y=element_blank())+
                   ggtitle(colnames(cell)[x]))
  do.call(grid.arrange,  plots)
}


```

```{r high MLLT1}
# high MLLT1

# Load the dataset
exp  <- high_exp
sig <- sig


# Run the function

high_proportion <- imsig(exp,sig)

plot_abundance (high_proportion)

# heatmap

high_proportion_transposed = t(high_proportion)

mat <- data.matrix(high_proportion_transposed)

col.pan <- colorpanel(100, "green", "black", "red")

heatmap.2(mat, col=col.pan, Rowv=FALSE, Colv=FALSE, scale="column", trace="none", dendrogram="none", cexRow=0.9, cexCol=0.1, density.info="none")

```

```{r low MLLT1}
# low MLLT1

# Load the dataset
exp  <- low_exp
sig <- sig


# Run the function

low_proportion <- imsig(exp,sig)

plot_abundance (low_proportion)

# heatmap

low_proportion_transposed = t(low_proportion)

mat <- data.matrix(low_proportion_transposed)

col.pan <- colorpanel(100, "green", "black", "red")

heatmap.2(mat, col=col.pan, Rowv=FALSE, Colv=FALSE, scale="column", trace="none", dendrogram="none", cexRow=0.9, cexCol=0.1, density.info="none")

```
