---
title: "pca_genes_analysis"
output:
  word_document: default
  html_document: default
---

```{r setup, include=FALSE, echo=FALSE}
knitr::opts_chunk$set

library(limma)
library(clusterProfiler)
library(org.Hs.eg.db)
library(pathview)
library(fgsea)
library(readr)
library(tidyverse)
```

# Downstream Analysis of MLLT1 Knockout RNA-Seq

Update: 28th April 2020

Pulled out list of genes that show most variation associated with PC1 and PC2 (from PCA plot)

## MCF-7

### Read in data
```{r}
# Genes from Hallmark gene sets (leading edge genes)

pca_genes <- read.csv("mcf_pca_genes.csv", header=TRUE)

# Differentially expressed genes (all)

mcf1_res <- read.csv("MCF1_res.csv", header=TRUE)
mcf2_res <- read.csv("MCF2_res.csv", header=TRUE)


# Add log2 fold change values

pca_mcf1 <- mcf1_res[is.element(mcf1_res$Row.names, pca_genes$ensembl),]
pca_mcf2 <- mcf2_res[is.element(mcf2_res$Row.names, pca_genes$ensembl),]


# Convert gene symbols to Entrez ID
pca_mcf1$Entrez.Gene <- mapIds(org.Hs.eg.db, keys=as.character(pca_mcf1$SYMBOL), keytype="SYMBOL", column="ENTREZID")
pca_mcf2$Entrez.Gene <- mapIds(org.Hs.eg.db, keys=as.character(pca_mcf2$SYMBOL), keytype="SYMBOL", column="ENTREZID")


```



### MCF-7 Variable Genes
```{r}

mcf_geneset <- as.character(pca_mcf1$Entrez.Gene)

mcf_ego <- enrichGO(gene = mcf_geneset, 
                universe = NULL, #all available genes in database
                OrgDb = org.Hs.eg.db, #Hs: homo sapiens
                ont ="BP", #molecular function, biological process, cellular component
                pAdjustMethod = "BH",
                pvalueCutoff = 0.01,
                qvalueCutoff = 0.05,  #q value is FDR adjusted p value
                readable = TRUE) #will show gene symbol in images later rather than Entrez Gene ID
# dimensions - number of GO terms
dim(mcf_ego)

# export GO information

mcf_go <- data.frame(mcf_ego$ID, mcf_ego$Description, mcf_ego$p.adjust, mcf_ego$geneID)
write.csv(mcf_go, file="mcf_go_bp.csv")

# plot results

barplot(mcf_ego)

```

```{r, fig.height=10, fig.width=12}

# logFC is optional, but if available will colour-code the nodes - MCF-7 Clone 1 log2 fold changes
mcf1_lfc <- pca_mcf1$log2FoldChange
# set name of object
names(mcf1_lfc) <- pca_mcf1$Entrez.Gene
head(mcf1_lfc)
cnetplot(mcf_ego, foldChange = mcf1_lfc, showCategory = 5, circular = FALSE)
# show 5 GO terms, if set circular to TRUE then GO terms won't be in middle, will be on edge of circle
# size = number of genes, fold change = red for upregulated, green for downregulated

# logFC is optional, but if available will colour-code the nodes - MCF-7 Clone 2 log2 fold changes
mcf2_lfc <- pca_mcf2$log2FoldChange
# set name of object
names(mcf2_lfc) <- pca_mcf2$Entrez.Gene
head(mcf2_lfc)
cnetplot(mcf_ego, foldChange = mcf2_lfc, showCategory = 5, circular = FALSE)
# show 5 GO terms, if set circular to TRUE then GO terms won't be in middle, will be on edge of circle
# size = number of genes, fold change = red for upregulated, green for downregulated

```

```{r, fig.height=3, fig.width=15}

# heat map 

heatplot(mcf_ego, foldChange = mcf1_lfc, showCategory = 4)
heatplot(mcf_ego, foldChange = mcf2_lfc, showCategory = 4)

```

## SUM159

### Read in data
```{r}
# Genes from Hallmark gene sets (leading edge genes)

pca_genes <- read.csv("sum_pca_genes.csv", header=TRUE)

# Differentially expressed genes (all)

sum1_res <- read.csv("SUM1_res.csv", header=TRUE)
sum2_res <- read.csv("SUM2_res.csv", header=TRUE)


# Add log2 fold change values

pca_sum1 <- sum1_res[is.element(sum1_res$Row.names, pca_genes$ensembl),]
pca_sum2 <- sum2_res[is.element(sum2_res$Row.names, pca_genes$ensembl),]


# Convert gene symbols to Entrez ID
pca_sum1$Entrez.Gene <- mapIds(org.Hs.eg.db, keys=as.character(pca_sum1$SYMBOL), keytype="SYMBOL", column="ENTREZID")
pca_sum2$Entrez.Gene <- mapIds(org.Hs.eg.db, keys=as.character(pca_sum2$SYMBOL), keytype="SYMBOL", column="ENTREZID")


```



### SUM159 Variable Genes
```{r}

sum_geneset <- as.character(pca_sum1$Entrez.Gene)

sum_ego <- enrichGO(gene = sum_geneset, 
                universe = NULL, #all available genes in database
                OrgDb = org.Hs.eg.db, #Hs: homo sapiens
                ont ="BP", #molecular function, biological process, cellular component
                pAdjustMethod = "BH",
                pvalueCutoff = 0.01,
                qvalueCutoff = 0.05,  #q value is FDR adjusted p value
                readable = TRUE) #will show gene symbol in images later rather than Entrez Gene ID
# dimensions - number of GO terms
dim(sum_ego)

sum_go <- data.frame(sum_ego$ID, sum_ego$Description, sum_ego$p.adjust, sum_ego$geneID)
write.csv(sum_go, file="sum_go_bp.csv")

# plot go terms
barplot(sum_ego)

```

```{r, fig.height=10, fig.width=12}

# logFC is optional, but if available will colour-code the nodes - SUM159 Clone 1 log2 fold changes
sum1_lfc <- pca_sum1$log2FoldChange
# set name of object
names(sum1_lfc) <- pca_sum1$Entrez.Gene
head(sum1_lfc)
cnetplot(sum_ego, foldChange = sum1_lfc, showCategory = 5, circular = FALSE)
# show 5 GO terms, if set circular to TRUE then GO terms won't be in middle, will be on edge of circle
# size = number of genes, fold change = red for upregulated, green for downregulated

# logFC is optional, but if available will colour-code the nodes - SUM159 Clone 2 log2 fold changes
sum2_lfc <- pca_sum2$log2FoldChange
# set name of object
names(sum2_lfc) <- pca_sum2$Entrez.Gene
head(sum2_lfc)
cnetplot(sum_ego, foldChange = sum2_lfc, showCategory = 5, circular = FALSE)
# show 5 GO terms, if set circular to TRUE then GO terms won't be in middle, will be on edge of circle
# size = number of genes, fold change = red for upregulated, green for downregulated

```

```{r, fig.height=3, fig.width=15}

# heat map 

heatplot(sum_ego, foldChange = sum1_lfc, showCategory=4)
heatplot(sum_ego, foldChange = sum2_lfc, showCategory=4)

```
